servers/src/main/java/tachyon/master/Master.java

```diff
@@ -74,4 +75,13 @@
    * @throws IOException
    */
   void stop() throws IOException;
+
+  /**
+   * Provides the master with a {@link Journal} capable of writing, in preparation to starting as
+   * the leader. This enables transitioning from standby master to leader master without having to
+   * construct a new master object.
+   *
+   * @param journal the {@link Journal} capable of writing
+   */
+  void upgradeToWriteJournal(Journal journal);
```

Comment:
That is a good point. How should that work? If the less functional read-only version is the base class, how do users of the base class know to get the write functionality? Does it have to do instance checks and casting?

Another option might be to have a journal interface, and both readonly and readwrite implement it.